packages/flutter/test/widgets/semantics_tester.dart

```diff
@@ -600,6 +602,9 @@ class _HasSemantics extends Matcher {
     if (!doesMatch) {
       matchState['would-match'] = item.generateTestSemanticsExpressionForCurrentSemanticsTree(childOrder);
     }
+    if (item.tester.binding.pipelineOwner.semanticsOwner == null) {
```

Comment:
nit: these are the kind of changes that I feel will be better to be kept in a different PR, as it least for me it requires a major context switch from the current PR when reviewing (+ I'd imagine the commit message for this line will make more sense)

Reply:
Ah, interesting. Ok. I included them because they were changes I needed to make to write the tests. Happy to split them out if you'd prefer.